### PR TITLE
Better typography

### PR DESCRIPTION
**Pull request by DaiQiangReal**
[中文模板 / Chinese Template](https://github.com/DouyinFE/semi-design/blob/main/.github/PULL_REQUEST_TEMPLATE.zh-CN.md)

- [X] I have read and followed [Pull Request Guidelines](https://github.com/DouyinFE/semi-design/blob/main/CONTRIBUTING-en-US.md#pull-request-guidelines) of the contributing guide.

### What kind of change does this PR introduce? (check at least one)

 - [X] Bugfix
 - [ ] Feature
 - [ ] Code style update
 - [ ] Refactor
 - [ ] Test Case
 - [ ] TypeScript definition update
 - [ ] Document improve
 - [ ] CI/CD improve
 - [ ] Branch sync
 - [ ] Other, please describe:

### PR description

Fixes #

### Changelog
🇨🇳 Chinese
- Perf: 提升 Typography 开启 Ellipsis 下 性能，减少 render 和计算次数 

---

🇺🇸 English
- Perf: Improve the performance of Typography when Ellipsis is turned on, and reduce the number of renders and calculations

### Checklist
- [ ] Test or no need
- [ ] Document or no need
- [ ] Changelog or no need

### Other
- [ ] Skip Changelog

### CODE REVIEW

File: packages/semi-ui/typography/base.tsx [hunk 1]
```diff
@@ -501,14 +525,14 @@ export default class Base extends Component<BaseTypographyProps, BaseTypographyS
         const { isTruncated, expanded, ellipsisContent } = this.state;
         if (expanded || !isTruncated) {
             return (
-                <>
+                <span onMouseEnter={this.onHover}>
```
Comment (YyumeiZhang) [inline packages/semi-ui/typography/base.tsx:528]: 这个地方不需要加 onMouseEnter，在 expanded 对应当前是展开状态， !isTruncated表示当前此处为非截断状态。这两个状态都属于 js 截断下，完全展示 props中的 children，这个位置不需要调用 this.onHover考虑是否hover需要展示tooltip？

**Comment by marshcat0** [comment]
这个MR这么久没有合并，是有啥问题吗？

**Comment by marshcat0** [comment]
> https://bytedance.larkoffice.com/docx/EEHKdE6AUoEKNdxA0EXcRumSnmg

我没有权限😂

**Comment by pointhalo** [comment]
image

单测跑不通，resizeable polyfill的问题

File: packages/semi-ui/typography/base.tsx [hunk 2]
```diff
@@ -194,7 +195,7 @@ export default class Base extends Component<BaseTypographyProps, BaseTypographyS
 
     componentDidMount() {
         if (this.props.ellipsis) {
-            this.onResize();
+            this.onResize().then(()=>runAfterTicks(()=>this.observerTakingEffect = true, 1));
```
Comment (pointhalo) [inline packages/semi-ui/typography/base.tsx:198]: 补一个注释。

**Pull request merged**